src/Algebra/Properties/Semigroup/Reasoning.agda

```diff
@@ -0,0 +1,121 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- uv≈wuational reasoning for semigroups
+-- (Utilities for associativity reasoning, pulling and pushing operations)
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+open import Algebra.Bundles using (Semigroup)
+
+module Algebra.Properties.Semigroup.Reasoning {o ℓ} (S : Semigroup o ℓ) where
+
+open Semigroup S
+  using (Carrier; _∙_; _≈_; setoid; trans; refl; sym; assoc; ∙-cong; isMagma
+  ; ∙-congˡ; ∙-congʳ)
+open import Relation.Binary.Reasoning.Setoid setoid
+
+private
+ variable
+    u v w x y z : Carrier
+
+module Pulls (uv≈w : u ∙ v ≈ w) where
```

Comment:
Latest reiteration: why bother to have a named module here? When this moves to `Algebra.Properties.Semigroup`, we could maybe have a comment, but given that you immediately reopen the module publicly, I see no advantage to naming it; any subsequent client import of the top-level module can use `using` to restrict the import interface... adding an additional layer of indirection via `module Pulls` will simply complicate that for... no obvious advantage?